### ISSUE TITLE
gcc10 Build break: fccanalyses

### ISSUE THREAD

**Issue by joequant** [key4hep/key4hep-spack#194]
!!!IMPORTANT!!!: To facilitate faster and easier response to your issue please provide in addition to the description of the issue also the following information

 - OS version: mageia9-broadwell
 - Compiler version: gcc 10.2
 - Package version: tag or commit ID, or GitHub branch: HEAD 20210408
 - Reproduced by: exact steps to reproduce the problem: checkout, setup environment (ROOT versions, LCG release), cmake, build, run...
 - Input: link to input files if applicable
 - Output: full build and run log output
 - Goal: A short description of what you are trying to achieve

Full build at 

http://dev1.bitquant.com.hk:8011/#/builders/3/builds/1332/steps/2/logs/stdio

```
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Utilities/TypeTraits.hpp:97:7:   required by substitution of 'template<template<class ...> class Op, class ... Args> using is_detected = typename Acts::detail::detector<Acts::detail::nonesuch, void, Op, Args ...>::value_t [with Op = Acts::Concepts::Linearizer::linTrack_t<const Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >, Acts::Result<Acts::LinearizedTrack>, const Acts::SingleBoundTrackParameters<Acts::SinglyCharged>&, const Eigen::Matrix<double, 4, 1, 0, 4, 1>&, const Acts::ContextType&, const Acts::ContextType&, Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >::State&>::fptr_meta_t; Args = {const Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField, Acts::detail::VoidAuctioneer>, Acts::detail::VoidNavigator>, Acts::PropagatorOptions<Acts::ActionList<>, Acts::AbortList<> > >, const Acts::SingleBoundTrackParameters<Acts::SinglyCharged>&, const Eigen::Matrix<double, 4, 1, 0, 4, 1>&, const Acts::ContextType&, const Acts::ContextType&, Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >::State&}]'
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Vertexing/LinearizerConcept.hpp:28:1:   required from 'constexpr const bool Acts::Concepts::Linearizer::linTrack_t<const Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >, Acts::Result<Acts::LinearizedTrack>, const Acts::SingleBoundTrackParameters<Acts::SinglyCharged>&, const Eigen::Matrix<double, 4, 1, 0, 4, 1>&, const Acts::ContextType&, const Acts::ContextType&, Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >::State&>::tv<const Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > > >::value'
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Utilities/TypeTraits.hpp:207:68:   required from 'constexpr const bool Acts::Concepts::has_method<const Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >, Acts::Result<Acts::LinearizedTrack>, template<class T, class R, class ... Arguments> struct Acts::Concepts::Linearizer::linTrack_t, const Acts::SingleBoundTrackParameters<Acts::SinglyCharged>&, const Eigen::Matrix<double, 4, 1, 0, 4, 1>&, const Acts::ContextType&, const Acts::ContextType&, Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > >::State&>'
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Vertexing/LinearizerConcept.hpp:34:50:   required from 'constexpr const bool Acts::Concepts::Linearizer::LinearizerConcept<Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > > >::linTrack_exists'
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Vertexing/LinearizerConcept.hpp:41:23:   required from 'struct Acts::Concepts::Linearizer::LinearizerConcept<Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > > >'
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Vertexing/LinearizerConcept.hpp:59:61:   required from 'constexpr const bool Acts::LinearizerConcept<Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > > >'
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Vertexing/AdaptiveMultiVertexFitter.hpp:40:17:   required from 'class Acts::AdaptiveMultiVertexFitter<Acts::SingleBoundTrackParameters<Acts::SinglyCharged>, Acts::HelicalTrackLinearizer<Acts::Propagator<Acts::EigenStepper<Acts::ConstantBField> > > >'
/tmp/user/spack-stage/spack-stage-fccanalyses-0.3.1-43z4omsazoijogaxm2rdmrk2ykjifaeg/spack-src/analyzers/dataframe/VertexFinderActs.cc:68:9:   required from here
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:120:33: error: 'class Acts::ConstantBField' has no member named 'validExtensionForStep'
  120 |   if (!state.stepping.extension.validExtensionForStep(state, *this) ||
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:121:33: error: 'class Acts::ConstantBField' has no member named 'k1'
  121 |       !state.stepping.extension.k1(state, *this, sd.k1, sd.B_first, sd.kQoP)) {
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~^~
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:137:35: error: 'class Acts::ConstantBField' has no member named 'k2'
  137 |     if (!state.stepping.extension.k2(state, *this, sd.k2, sd.B_middle, sd.kQoP,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:143:35: error: 'class Acts::ConstantBField' has no member named 'k3'
  143 |     if (!state.stepping.extension.k3(state, *this, sd.k3, sd.B_middle, sd.kQoP,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:151:35: error: 'class Acts::ConstantBField' has no member named 'k4'
  151 |     if (!state.stepping.extension.k4(state, *this, sd.k4, sd.B_last, sd.kQoP, h,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:201:35: error: 'class Acts::ConstantBField' has no member named 'finalize'
  201 |     if (!state.stepping.extension.finalize(state, *this, h, D)) {
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/opt/spack/linux-mageia9-broadwell/gcc-10.2.0/acts-6.00.0-kelkqnuk2n7bq64ljbtttnrhnygro4to/include/Acts/Propagator/EigenStepper.ipp:208:35: error: 'class Acts::ConstantBField' has no member named 'finalize'
  208 |     if (!state.stepping.extension.finalize(state, *this, h)) {
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
make[2]: *** [analyzers/dataframe/CMakeFiles/FCCAnalyses.dir/build.make:224: analyzers/dataframe/CMakeFiles/FCCAnalyses.dir/VertexFitterActs.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [analyzers/dataframe/CMakeFiles/FCCAnalyses.dir/build.make:210: analyzers/dataframe/CMakeFiles/FCCAnalyses.dir/VertexFinderActs.cc.o] Error 1
make[2]: Leaving directory '/tmp/user/spack-stage/spack-stage-fccanalyses-0.3.1-43z4omsazoijogaxm2rdmrk2ykjifaeg/spack-build-43z4oms'
make[1]: *** [CMakeFiles/Makefile2:102: analyzers/dataframe/CMakeFiles/FCCAnalyses.dir/all] Error 2
make[1]: Leaving directory '/tmp/user/spack-stage/spack-stage-fccanalyses-0.3.1-43z4omsazoijogaxm2rdmrk2ykjifaeg/spack-build-43z4oms'
make: *** [Makefile:139: all] Error 2
[1;31m==>[0m Error: ProcessError: Command exited with st
```

**Comment by vvolkl** [comment]
I think this is a version mismatch, fccanalyses need acts v5, which needs to be added to the recipe as a requirement.

**Comment by vvolkl** [comment]
Fixed in #334
